.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,23 @@
+name: Golangci-lint
```

Comment:
We already have a workflow to do static analysis on of source code on each PR defined in `.github/workflow/build.yaml`
Instead of adding another YAML, could we run golangci-lint action as a step under the `verify-code` job? This way we'll group checks and workflow triggers in one place.